### PR TITLE
New: Kit Elements Defaults - Introduction [ED-8357]

### PR DESCRIPTION
**Pull request by Nevoss**

### CODE REVIEW

File: core/modules-manager.php [hunk 1]
```diff
@@ -98,7 +98,8 @@ public function get_modules_names() {
 			'web-cli',
 			'promotions',
 			'notes',
-			'performance-lab',
+      'performance-lab',
+			'kit-elements-defaults',
```
Comment (matipojo) [inline core/modules-manager.php:102]:
```suggestion
			'performance-lab',
			'kit-elements-defaults',
```

**Review by matipojo — CHANGES_REQUESTED** [review]

File: modules/kit-elements-defaults/assets/js/editor/commands/delete.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import { deleteElementDefaults } from '../api';
+
+export default class Delete extends $e.modules.CommandBase {
+	async apply( { type } ) {
+		$e.internal( 'panel/state-loading' );
+
+		try {
+			await deleteElementDefaults( type );
+		} catch ( e ) {
+			// TODO: Show error toast.
```
[the remark this answers is not in the document]
Comment (Nevoss) [reply modules/kit-elements-defaults/assets/js/editor/commands/delete.js:10]: Done

File: modules/kit-elements-defaults/assets/js/editor/commands/create.js [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+		[ '__dynamic__', '__globals__' ].forEach( ( type ) => {
+			const specialSettings = this.extractSpecialSettings( type, container );
+
+			if ( ! isPopulatedObject( specialSettings ) ) {
+				return;
+			}
+
+			settings[ type ] = specialSettings;
+		} );
```
[the remark this answers is not in the document]
Comment (Nevoss) [reply modules/kit-elements-defaults/assets/js/editor/commands/create.js:24]: Done

File: modules/kit-elements-defaults/assets/js/editor/commands/create.js [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+			specialSettings = localSettings?.[ type ] || {};
+
+		const entries = Object.entries( specialSettings )
+			// Remove controls that have local value.
+			.filter( ( [ settingsName ] ) => ! Object.prototype.hasOwnProperty.call( localSettings, settingsName ) )
+			// Remove controls that are not exist.
+			.filter( ( [ settingName ] ) => !! settings.controls[ settingName ] );
+
+		return Object.fromEntries( entries );
+	}
```
[the remark this answers is not in the document]
Comment (Nevoss) [reply modules/kit-elements-defaults/assets/js/editor/commands/create.js:66]: Done

File: modules/kit-elements-defaults/utils/settings-sanitizer.php [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+	public function sanitize( $settings, $type ) {
+		$element = $this->get_element_instance( $type );
+
+		if ( ! $element ) {
+			return [];
+		}
+
+		$regular_settings = $this->remove_invalid_settings( $element, $settings );
+		$special_settings = $this->remove_invalid_special_settings( $element, $settings );
+
```
[the remark this answers is not in the document]
Comment (Nevoss) [reply modules/kit-elements-defaults/utils/settings-sanitizer.php:36]: Done

**Review by matipojo — APPROVED** [review]

**Pull request merged**